### PR TITLE
consensus/pbft: add unit tests for verifyPrepare

### PR DESCRIPTION
**Pull request by bailantaotao**

### CODE REVIEW

File: consensus/pbft/core/core_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+			if x == y {
+				continue
+			}
+			x.sys.backends[i].AddPeer(fmt.Sprintf("%d", y.sys.backends[j].ID()), &y.sys.backends[j].privateKey.PublicKey)
+		}
+	}
+}
+
+func TestCore_NewRequest(t *testing.T) {
```
Comment (alanchchen) [inline consensus/pbft/core/core_test.go]: According to go-ethereum's naming convention, I suggest changing it to `TestNewRequest`

File: consensus/pbft/core/prepare_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/consensus/pbft"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// view number is not checked for now
+func TestCore_verifyPrepare(t *testing.T) {
```
Comment (alanchchen) [inline consensus/pbft/core/prepare_test.go]: `TestVerifyPrepare`

File: consensus/pbft/core/core_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+package core
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func connectAll(bs []*testSystemBackend) {
```
Comment (alanchchen) [inline consensus/pbft/core/core_test.go]: Is this necessary? `AddPeer` does nothing but print message only.

**Comment by alanchchen** [comment]
Please revise your commit messages as well. Use the correct package name.

File: consensus/pbft/core/core_test.go (hunk 1 above)
Comment (bailantaotao) [inline consensus/pbft/core/core_test.go]: Ok, the naming is recommended by IDE. XDD

File: consensus/pbft/core/core_test.go (hunk 3 above)
Comment (bailantaotao) [inline consensus/pbft/core/core_test.go]: 原本想法是在跑integration tests的時候可以知道該test共有哪些validators
但經你這麼一說我也覺得好像不是很重要了ＸＤ
然後跑其他tests時，該log反而有點礙眼

我會把它拿掉

Comment (bailantaotao) [inline consensus/pbft/core/core_test.go]: fixed in https://github.com/getamis/go-ethereum/pull/25/commits/bf08f3378572de0b828cf1e05f74c6224654d750

File: consensus/pbft/core/core_test.go (hunk 1 above)
Comment (bailantaotao) [inline consensus/pbft/core/core_test.go]: fixed in https://github.com/getamis/go-ethereum/pull/25/commits/bf08f3378572de0b828cf1e05f74c6224654d750

File: consensus/pbft/core/prepare_test.go (hunk 2 above)
Comment (bailantaotao) [inline consensus/pbft/core/prepare_test.go]: fixed in https://github.com/getamis/go-ethereum/pull/25/commits/bf08f3378572de0b828cf1e05f74c6224654d750

File: consensus/pbft/core/prepare_test.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+			expected: pbft.ErrSubjectNotMatched,
+			prepare: &pbft.Subject{
+				&pbft.View{ViewNumber: big.NewInt(1), Sequence: big.NewInt(0)},
+				[]byte{1},
+			},
+			self: &pbft.Subject{
+				&pbft.View{ViewNumber: big.NewInt(0), Sequence: big.NewInt(0)},
+				[]byte{1},
+			},
+		},
```
Comment (tailingchen) [inline consensus/pbft/core/prepare_test.go]: Do we need to test prepare message with same view number but different sequence number?

Comment (bailantaotao) [inline consensus/pbft/core/prepare_test.go]: 我認爲不需要，前面檢查是否為old message之後
還會再確認內容是否相符合，我可以加個註記
```
func (c *core) verifyPrepare(prepare *pbft.Subject, src *pbft.Validator) error {
	logger := c.logger.New("from", src.ID(), "state", c.state)

	if prepare.View.Sequence != nil &&
		c.subject != nil &&
		prepare.View.Sequence.Cmp(c.subject.View.Sequence) < 0 {
		logger.Warn("Old message", "expected", c.subject, "got", prepare)
		return pbft.ErrOldMessage
	}

	if !reflect.DeepEqual(prepare, c.subject) {
		logger.Warn("Subject not match", "expected", c.subject, "got", prepare)
		return pbft.ErrSubjectNotMatched
	}

	return nil
}
```

**Pull request merged**